### ISSUE TITLE
Feat: Addition of a Tracker.resetTracking() call?

### ISSUE THREAD

**Issue by d4rken** [matomo-org/matomo-sdk-android#261]
Also see #259 

"Old user logs out, new user logs in, track as new visit" seems to be a common yet problematic use-case that could be made easier by providing a call like:

```java
class Tracker {
   /* .... */ 
   void resetVisit() {
        // Null user id
        // Set new random visitor id
        // set NEW_VISIT = 1
    }
}
```

Apparently the JavaScript SDK has this, I'm not certain whether the iOS SDK has this?

`resetVisit` might be a bit missleading and something like `resetTracking` would be more appropriate in conveying the effect.

**Comment by mattab** [comment]
Fyi here is how we now recommend to do this in JavaScript: https://developer.matomo.org/guides/tracking-javascript-guide#when-user-logs-out-reset-user-id And similarly documented in https://matomo.org/faq/how-to/faq_187/

it's a bit convoluted/complex but that's the best we got :)

**Comment by VikasRao** [comment]
> https://matomo.org/faq/how-to/faq_187/

but how to push new_visit=1 from android, should we use Tracker.setUserId(1) or setVisitorId  or any other method?

**Comment by d4rken** [comment]
```java
        TrackMe trackMe = TrackHelper.track()
                .event("exampleCategory","exampleAction")
                .build();
        trackMe.set(QueryParams.SESSION_START, 1);
        mTracker.track(trackMe);
```

**Comment by hannesa2** [comment]
Solved with  #373
If not, please reopen it
